streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -229,22 +240,24 @@ private void updateProcessorContext(final StampedRecord record, final ProcessorN
 
     /**
      * <pre>
-     *  - flush state and producer
-     *  - write checkpoint
-     *  - commit offsets
+     * - flush state and producer
+     * - if(!eos) write checkpoint
+     * - commit offsets and maybe start new transaction
      * </pre>
      */
     @Override
-    public void commit() {
+    public void commit(final boolean startNewTransaction) {
```

Comment:
I understand, what I meant is to have a `commit(eosEnabled: Boolean)`, and the stream thread can then call it with `commit(eosEnabled)`, while its own `suspend` call `commit(false)`. It's just that the parameter semantics would be more general than `startNewTransaction`, which should logically be agnostic from the thread's point of view but only wraps in the task class itself.